bindings/python/test/test_bson.py

```diff
@@ -22,34 +22,73 @@
 from pymongoarrow.types import int32, int64
 
 
-class TestBSONUtil(TestCase):
-    @classmethod
-    def setUpClass(cls):
+class TestBsonToArrowConversionBase(TestCase):
+    def setUp(self):
+        self.schema = Schema({'_id': int32(),
+                              'data': int64()})
+        self.context = PyMongoArrowContext.from_schema(
+            self.schema)
+
+    @staticmethod
+    def _generate_payload(doclist):
+        payload = b''
+        for doc in doclist:
+            payload += encode(doc)
+        return payload
+
+    def _run_test(self, doclist, as_dict):
+        payload = type(self)._generate_payload(doclist)
+
+        process_bson_stream(payload, self.context)
+        table = self.context.finish()
+        table_dict = table.to_pydict()
+
+        for key, value in table_dict.items():
+            self.assertEqual(value, as_dict[key])
+
+
+class TestValidBsonToArrowConversion(TestBsonToArrowConversionBase):
+    def test_simple(self):
         docs = [{'_id': 1, 'data': 10},
                 {'_id': 2, 'data': 20},
                 {'_id': 3, 'data': 30},
                 {'_id': 4, 'data': 40}]
+        as_dict = {
+            '_id': [1, 2, 3, 4],
+            'data': [10, 20, 30, 40]}
 
-        payload = b''
-        for doc in docs:
-            payload += encode(doc)
+        self._run_test(docs, as_dict)
 
-        as_dict = defaultdict(list)
-        for doc in docs:
-            for key, value in doc.items():
-                as_dict[key].append(value)
+    def test_with_nulls(self):
+        docs = [{'_id': 1, 'data': 10},
+                {'_id': 2, 'data': 20},
+                {'_id': 3},
+                {'_id': 4, 'data': 40},
+                {'foo': 1}]
+        as_dict = {
+            '_id': [1, 2, 3, 4, None],
+            'data': [10, 20, None, 40, None]}
+
+        self._run_test(docs, as_dict)
 
-        cls.as_dict = as_dict
-        cls.docs = docs
-        cls.schema = Schema({'_id': int32(),
-                             'data': int64()})
-        cls.bson_stream = payload
+
+class TestInvalidBsonToArrowConversion(TestBsonToArrowConversionBase):
+    @staticmethod
+    def _generate_payload(doclist):
+        payload = b''
+        for doc in doclist:
+            payload += encode(doc)
+        return payload[:-2]
```

Comment:
Can this be replaced with `super()._generate_payload()[:-2]`?

Reply:
Doesn't seem to work.. Some strange interaction with `super` and `staticmethod` I think?
```
======================================================================
ERROR: test_simple (test.test_bson.TestInvalidBsonToArrowConversion)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/pmital/Developer/mongo-arrow-github/bindings/python/test/test_bson.py", line 95, in test_simple
    self._run_test(docs, as_dict)
  File "/Users/pmital/Developer/mongo-arrow-github/bindings/python/test/test_bson.py", line 40, in _run_test
    payload = type(self)._generate_payload(doclist)
  File "/Users/pmital/Developer/mongo-arrow-github/bindings/python/test/test_bson.py", line 78, in _generate_payload
    return super()._generate_payload()[:-2]
TypeError: super(type, obj): obj must be an instance or subtype of type

----------------------------------------------------------------------
```

Reply:
Apparently you need to do:
```
super(TestInvalidBsonToArrowConversion, TestInvalidBsonToArrowConversion)._generate_payload()[:-2]?
```
https://stackoverflow.com/a/26807879/8637283

Reply:
Done.